clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/task/datawriter/AbstractPartitionWriter.java

```diff
@@ -102,12 +104,22 @@ public VeniceWriterMessage(
         PubSubProducerCallback callback,
         boolean enableWriteCompute,
         int derivedValueSchemaId) {
-      this(keyBytes, valueBytes, valueSchemaId, -1, null, callback, enableWriteCompute, derivedValueSchemaId);
+      this(
+          keyBytes,
+          valueBytes,
+          APP_DEFAULT_LOGICAL_TS,
+          valueSchemaId,
+          -1,
+          null,
+          callback,
+          enableWriteCompute,
+          derivedValueSchemaId);
     }
 
     public VeniceWriterMessage(
         byte[] keyBytes,
         byte[] valueBytes,
+        long topLevelLogicalts,
```

Comment:
In many places, we're using `Long` to store the value, and we cast it to `long` and use `<0` as the deciding criteria. If we plan to use `-1` default value, then we don't need to use `Long` at all.